### ISSUE TITLE
Improve logging

### ISSUE THREAD

**Issue by ivarconr** [Unleash/unleash#175]
Logging requires more customization. E.g. we need a different format when we want kibana / elastic search to index and show the logs. 

Consider using [bole](https://www.npmjs.com/package/bole) as log implementation and provide hooks to customize it when using unleash as a lib.

**Comment by SimenB** [comment]
Could you make it into an option? That way we can pass our own logger.

If you take a function returning an object with `.warn`, `.error` etc, it matches the API of bole.

```js
function loggerFactory (loggerName) {
  return {
    info: () => {},
    warn: () => {},
    error: () => {},
    debug: () => {},
    trace: () => {},
  };
}
```

Something like that would be the contract

**Comment by ivarconr** [comment]
Sure. Error should probably also accept an optional error object?

**Comment by SimenB** [comment]
Yeah. Our logger differs from bole there, though. They have the error as the first arg, we have as second.
Should change ours, I suppose

**Comment by ivarconr** [comment]
> Log methods accept Error objects and print appropriate Error properties, including a full stack trace (including any cause where supported)

**Comment by ivarconr** [comment]
solved via #258
